frontend/src/plugins/impl/DataTablePlugin.tsx

```diff
@@ -508,20 +516,44 @@ const DataTableComponent = ({
   );
 
   const rowSelection = useMemo(
-    () => Object.fromEntries((value || []).map((v) => [v, true])),
+    () => Object.fromEntries((value || []).map((v) => [v.rowId, true])),
     [value],
   );
 
   const handleRowSelectionChange: OnChangeFn<RowSelectionState> = useEvent(
     (updater) => {
       if (selection === "single") {
         const nextValue = Functions.asUpdater(updater)({});
-        setValue(Object.keys(nextValue).slice(0, 1));
+        setValue(
+          Object.keys(nextValue)
+            .slice(0, 1)
+            .map((r) => ({ rowId: r })),
```

Comment:
i think it would be preferable to keep this as a list instead of paying the cost to transform it to a dictionary and back